lib/logsource.c

```diff
@@ -574,6 +575,25 @@ log_source_override_program(LogSource *self, LogMessage *msg)
   log_msg_set_value(msg, LM_V_PROGRAM, self->options->program_override, self->options->program_override_len);
 }
 
+static gchar *
+_get_pid_string(void)
+{
+#define MAX_PID_CHAR_COUNT 20 /* max PID on 64 bit systems is 2^64 - 1, which is 19 characters, +1 for terminating 0 */
+
+  static gchar pid_string[MAX_PID_CHAR_COUNT];
+
+  if (pid_string[0] == '\0')
+    {
+#ifdef _WIN32
+      g_snprintf(pid_string, MAX_PID_CHAR_COUNT, "%lu", GetCurrentProcessId());
+#else
+      g_snprintf(pid_string, MAX_PID_CHAR_COUNT, "%d", getpid());
```

Comment:
64bit pids can be problematic with `%d` (the type is `pid_t`), but I've never seen such a thing IRL.

Reply:
AFAIK `pid_t` is still 32 bits.

Reply:
Yeah, it should be.
My pid_max is `4194304` with Linux 5.6.15.